bigbluebutton/bigbluebutton
# feat(plugins): Added support for persistEvent

### What does this PR do?

This PR implemented the `pluginApi.persistEvent` function just as requested by https://github.com/bigbluebutton/bigbluebutton-html-plugin-sdk/issues/121.


### Closes Issue(s)
Closes https://github.com/bigbluebutton/bigbluebutton-html-plugin-sdk/issues/121

### How to test
Create a meeting with `record=true` to capture `events.xml` and with the following parameter:
```
pluginManifests=[{"url":"http://172.17.0.1:4701/manifest.json"}]
```
run the sample-use-meeting plugin with an  `npm start` and with the correct SDK.

Wait for a few seconds, end the meeting and check the `events.xml` by doing the folloing commands:

```bash
sudo updatedb
vi $(locate events.xml | grep <meeting-id>)
```

If you are using the use-meeting sample, then, your `events.xml` should look something like:

```xml
  <event timestamp="25004947" module="PLUGIN" eventname="PluginGeneratedEvent">
    <payloadJson>{"foo":"bar"}</payloadJson>
    <pluginEventName>eventFromUseMeetingSample</pluginEventName>
    <date>2024-10-30T18:00:11.929Z</date>
    <pluginName>SampleUseMeeting</pluginName>
    <userId>w_sxlfjbcb0yxs</userId>
    <timestampUTC>1730311211929</timestampUTC>
  </event>
  <event timestamp="25004949" module="PLUGIN" eventname="PluginGeneratedEvent">
    <payloadJson>{"foo":"bar"}</payloadJson>
    <pluginEventName>eventFromUseMeetingSample</pluginEventName>
    <date>2024-10-30T18:00:11.931Z</date>
    <pluginName>SampleUseMeeting</pluginName>
    <userId>w_sxlfjbcb0yxs</userId>
    <timestampUTC>1730311211931</timestampUTC>
  </event>
```

The meeting id can be found with `/getMeetingInfo` endpoint in the ApiMate for instance.

### More
Closely related to https://github.com/bigbluebutton/bigbluebutton-html-plugin-sdk/pull/122 
Depends on https://github.com/bigbluebutton/bigbluebutton/pull/21368
